Deprecated pyqtconfig fails build on ArchLinux

Hello

I don't know anything about this project or python at all, but it won't build on ArchLinux: `Linux 4.7.1-1-ARCH x86_64 GNU/Linux` or any other (probably).

From: https://bugs.archlinux.org/task/40842
`configure-ng.py is the new build system, pyqtconfig.py is a product of the old (deprecated) build system.`

This is my log:

``` plain
-- Found PythonInterp: /usr/bin/python2.7 (found version "2.7.12") 
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found version "2.7.12") 
Traceback (most recent call last):
  File "/build/seexpr/src/SeExpr-2.11/src/build/build-info", line 14, in <module>
    from PyQt4 import pyqtconfig
ImportError: cannot import name pyqtconfig
Traceback (most recent call last):
  File "/build/seexpr/src/SeExpr-2.11/src/build/build-info", line 14, in <module>
    from PyQt4 import pyqtconfig
ImportError: cannot import name pyqtconfig
Traceback (most recent call last):
  File "/build/seexpr/src/SeExpr-2.11/src/build/build-info", line 14, in <module>
    from PyQt4 import pyqtconfig
ImportError: cannot import name pyqtconfig
Traceback (most recent call last):
  File "/build/seexpr/src/SeExpr-2.11/src/build/build-info", line 14, in <module>
    from PyQt4 import pyqtconfig
ImportError: cannot import name pyqtconfig

...

[ 77%] SeExprEditor.sip'
sip: Unable to find file "QtGui/QtGuimod.sip"
make[2]: *** [src/SeExprEditor/CMakeFiles/expreditor.dir/build.make:62: src/SeExprEditor/sipexpreditorpart0.cpp] Error 1
make[1]: *** [CMakeFiles/Makefile2:219: src/SeExprEditor/CMakeFiles/expreditor.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
```

--- a/src/build/build-info
+++ b/src/build/build-info
@@ -11,7 +11,6 @@ if 'MACOSX_DEPLOYMENT_TARGET' in os.environ:
 from distutils import sysconfig as sc
 
 import sipconfig
-from PyQt4 import pyqtconfig
 
 def main():
     parser = argparse.ArgumentParser()
@@ -56,6 +55,7 @@ def python_site(args):
 
 
 def pyqt4_sip(args):
+    from PyQt4 import pyqtconfig
     pkg_cfg = pyqtconfig._pkg_config
     return pkg_cfg['pyqt_sip_dir']
 

--- a/src/ui/CMakeLists.txt
+++ b/src/ui/CMakeLists.txt
@@ -28,7 +28,15 @@ endmacro()
 get_build_info(python-site PYTHON_SITE)
 get_build_info(python-inc PYTHON_INCLUDE_DIR)
 get_build_info(sip-inc SIP_INCLUDE_DIR)
-get_build_info(pyqt4-sip PYQT4_SIP)
+
+list(APPEND CMAKE_MODULE_PATH "/usr/share/apps/cmake/modules/") # needed for e.g Arch Linux as new versions of CMake only search a version-specific module path. Should be no problem if the path does not exist.
+find_package(PyQt4)
+set(PYQT4_SIP ${PYQT4_SIP_DIR})
+if(PYQT4_FOUND)
+
+else()
+  get_build_info(pyqt4-sip PYQT4_SIP)
+endif()
 
 set(CMAKE_INSTALL_PYTHON "${PYTHON_SITE}/SeExpr2" )
 

--- a/src/ui/CMakeLists.txt
+++ b/src/ui/CMakeLists.txt
@@ -29,13 +29,14 @@ get_build_info(python-site PYTHON_SITE)
 get_build_info(python-inc PYTHON_INCLUDE_DIR)
 get_build_info(sip-inc SIP_INCLUDE_DIR)
 
-list(APPEND CMAKE_MODULE_PATH "/usr/share/apps/cmake/modules/") # needed for e.g Arch Linux as new versions of CMake only search a version-specific module path. Should be no problem if the path does not exist.
+if (EXISTS "/usr/share/apps/cmake/modules")
+    # Needed for some versions of CMake, which only look in version-specific module path
+    list(APPEND CMAKE_MODULE_PATH "/usr/share/apps/cmake/modules")
+endif()
 find_package(PyQt4)
 set(PYQT4_SIP ${PYQT4_SIP_DIR})
-if(PYQT4_FOUND)
-
-else()
-  get_build_info(pyqt4-sip PYQT4_SIP)
+if (NOT PYQT4_FOUND)
+    get_build_info(pyqt4-sip PYQT4_SIP)
 endif()
 
 set(CMAKE_INSTALL_PYTHON "${PYTHON_SITE}/SeExpr2" )

